packages/core/src/workflows/workflow.ts

```diff
@@ -795,7 +805,7 @@ export class Workflow<
     steps: TStep | Workflow | (TStep | Workflow)[],
   ): Omit<WorkflowBuilder<this>, 'then' | 'after'> {
     const stepsArray = Array.isArray(steps) ? steps : [steps];
-    const stepKeys = stepsArray.map(step => this.#makeStepKey(step));
+    const stepKeys = stepsArray.map(step => (typeof step === 'string' ? step : this.#makeStepKey(step)));
```

Comment:
Should we move this logic inside #makeStepKey ?
```suggestion
    const stepKeys = stepsArray.map(step => (typeof step === 'string' ? step : this.#makeStepKey(step)));
```